core/server/master/src/test/java/alluxio/master/meta/UpdateCheckTest.java

```diff
@@ -46,86 +53,148 @@ public void before() {
+    Assert.assertTrue(userAgentString.equals("cluster1; gce"));
   }
 
   @Test
-  public void userAgentStringEC2AMI() throws Exception {
+  public void userAgentEnvironmentStringEC2AMI() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     // When no user data in this ec2, null is returned
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn(null);
 
-    String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; ProductCode:random123code; ec2)",
-            ProjectConstants.VERSION)));
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; ec2"));
   }
 
   @Test
-  public void userAgentStringEC2CFT() throws Exception {
+  public void userAgentEnvironmentStringEC2CFT() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     Mockito.when(EnvironmentUtils.isCFT(Mockito.anyString())).thenReturn(true);
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn("{ \"cft_configure\": {}}");
 
-    String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; ProductCode:random123code; cft; ec2)",
-            ProjectConstants.VERSION)));
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; cft; ec2"));
   }
 
   @Test
-  public void userAgentStringEC2EMR() throws Exception {
+  public void userAgentEnvironmentStringEC2EMR() throws Exception {
     Mockito.when(EnvironmentUtils.isEC2()).thenReturn(true);
     Mockito.when(EnvironmentUtils.getEC2ProductCode()).thenReturn("random123code");
     Mockito.when(EnvironmentUtils.isEMR(Mockito.anyString())).thenReturn(true);
     Mockito.when(EC2MetadataUtils.getUserData()).thenReturn("emr_apps");
 
+    String userAgentString = UpdateCheck.getUserAgentEnvironmentString("cluster1");
+    Assert.assertTrue(userAgentString.equals("cluster1; ProductCode:random123code; emr; ec2"));
+  }
+
+  @Test
+  public void featureStringEmbeddedJournal() {
+    ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    Assert.assertFalse(UpdateCheck.getFeatureString().contains("embedded"));
+    ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("embedded"));
+  }
+
+  @Test
+  public void featureStringRocks() {
+    ServerConfiguration.set(PropertyKey.MASTER_METASTORE, MetastoreType.ROCKS);
```

Comment:
i'm confused why you can use MetastoreType.ROCKS here in the test file but not the source file, is this expected?

Reply:
It's because of the dependency of different modules. Both `UpdateCheckTest` and  `JournalType" are under alluxio-core-server-common, while `MetastoreType` is under alluxio-core-server-master. If we want to reuse the enum MetastoreType, we have to make alluxio-core-server-master as a dependency as alluxio-core-server-common, which is not expected. The workaround is using string instead, and check consistency in a unit test.

Reply:
does the test module have different dependencies than the source module? because you are importing `import alluxio.master.metastore.MetastoreType;` in this test class. again, my confusion is why you can import it here in the test class but not in the utils class